server/src/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -1661,6 +1661,19 @@ private boolean upgradeRunningVirtualMachine(Long vmId, Long newServiceOfferingI
                     + ",memory=," + currentMemory + ")");
         }
 
+        _offeringDao.loadDetails(currentServiceOffering);
```

Comment:
For a bit of improvement, before loading, we can check if the value is already loaded then we can skip this, else load it. Something like this.

if(currentServiceOffering.getDetails() == null) {
_offeringDao.loadDetails(currentServiceOffering);
}

Same for newServiceOffering.